### PR TITLE
Expo scanner merged with RN scanner

### PR DESCRIPTION
**Pull request by godrei**

### CODE REVIEW

File: scanners/flutter/flutter.go [hunk 1]
```diff
@@ -86,6 +86,7 @@ func findProjectLocations(searchDir string) ([]string, error) {
 
 	filters := []utility.FilterFunc{
 		utility.BaseFilter("pubspec.yaml", true),
+		utility.ComponentFilter("node_modules", false),
```
Comment (BirmacherAkos) [inline scanners/flutter/flutter.go]: Nice catch 👍

File: scanners/reactnative/expo.go [hunk 2]
```diff
@@ -13,171 +12,36 @@ import (
...
-// Name ...
-const Name = "react-native-expo"
-
-// Scanner ...
-type Scanner struct {
-	searchDir      string
-	packageJSONPth string
-	usesExpoKit    bool
+// AppJSONError represents an issue with app.json file.
+type AppJSONError struct {
```
Comment (trapacska) [inline scanners/reactnative/expo.go]: Does it still need to be exported?

File: scanners/reactnative/expo.go [hunk 3]
```diff
@@ -13,171 +12,36 @@ import (
...
+	reason      string
+	explanation string
 }
 
-// NewScanner ...
-func NewScanner() *Scanner {
-	return &Scanner{}
+// Error implements the error interface
+func (e AppJSONError) Error() string {
+	return fmt.Sprintf("app.json file (%s) %s\n%s", e.appJSONPth, e.reason, e.explanation)
```
Comment (trapacska) [inline scanners/reactnative/expo.go]: actually for this reason I wouldn't implement a new type that is matching for the main error interface.
what about:
```
func appJSONError([..appJSONPth, reason, explanation.....]) error {
	return fmt.Errorf("app.json file (%s) %s\n%s", appJSONPth, reason, explanation)
}
```

Comment (trapacska) [reply scanners/reactnative/expo.go]: So at the end you can use almost the same way:
```
return models.OptionNode{}, warnings, appJSONError{appJSONPth, "missing expo entry", errorMessage}
```

File: _tests/integration/manual_config_test.go [hunk 4]
```diff
@@ -1619,10 +1650,10 @@ configs:
               - workdir: $WORKDIR
               - command: test
           - deploy-to-bitrise-io@%s: {}
-    default-react-native-expo-plain-config: |
+    default-react-native-expo-expo-kit-config: |
       format_version: "%s"
       default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
-      project_type: react-native-expo
+      project_type: default-react-native-expo-config
```
Comment (lpusok) [inline _tests/integration/manual_config_test.go]: Please check the project type.

File: _tests/integration/manual_config_test.go [hunk 5]
```diff
@@ -1529,11 +1565,10 @@ configs:
               inputs:
               - command: test
           - deploy-to-bitrise-io@%s: {}
-  react-native-expo:
-    default-react-native-expo-expo-kit-config: |
+    default-react-native-expo-config: |
       format_version: "%s"
       default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
-      project_type: react-native-expo
+      project_type: default-react-native-expo-config
```
Comment (lpusok) [inline _tests/integration/manual_config_test.go]: Project type seems invalid.

**Pull request merged**